ev3dev/control/GyroBalancer.py

```diff
@@ -1,398 +1,477 @@
-                ##  Computing Motor Speed
-                ###############################################################
-                motorAngularSpeed = (motorAngle - motorAngleHistory[0])/(motorAngleHistoryLength * loopTimeSec)
-                motorAngularSpeedError = motorAngularSpeed - motorAngularSpeedReference
-                motorAngleHistory.append(motorAngle)
+                # Platform specific constants and conversions
 
-                ###############################################################
-                ##  Computing the motor duty cycle value
-                ###############################################################
-                motorDutyCycle =(gainGyroAngle  * gyroEstimatedAngle
-                               + gainGyroRate   * gyroRate
-                               + gainMotorAngle * motorAngleError
-                               + gainMotorAngularSpeed * motorAngularSpeedError
-                               + gainMotorAngleErrorAccumulated * motorAngleErrorAccumulated)
+                # For the LEGO EV3 Gyro in Rate mode, 1 unit = 1 deg/s
+                degPerSecondPerRawGyroUnit = 1
 
-                ###############################################################
-                ##  Apply the signal to the motor, and add steering
-                ###############################################################
-                SetDuty(motorDutyCycleRight, motorDutyCycle + self.steering)
-                SetDuty(motorDutyCycleLeft, motorDutyCycle - self.steering)
+                # Express the above as the rate in rad/s per gyro unit
+                radiansPerSecondPerRawGyroUnit = degPerSecondPerRawGyroUnit *\
+                    radiansPerDegree
 
-                ###############################################################
-                ##  Update angle estimate and Gyro Offset Estimate
-                ###############################################################
-                gyroEstimatedAngle = gyroEstimatedAngle + gyroRate * loopTimeSec
-                gyroOffset = (1 - gyroDriftCompensationRate) * gyroOffset + gyroDriftCompensationRate * gyroRateRaw
+                # For the LEGO EV3 Large Motor 1 unit = 1 deg
+                degPerRawMotorUnit = 1
+
+                # Express the above as the angle in rad per motor unit
+                radiansPerRawMotorUnit = degPerRawMotorUnit * radiansPerDegree
+
+                # On the EV3, "1% speed" corresponds to 1.7 RPM (if speed
+                # control were enabled).
+                RPMperPerPercentSpeed = 1.7
+
+                # Convert this number to the speed in deg/s per "percent speed"
+                degPerSecPerPercentSpeed = RPMperPerPercentSpeed * 360 / 60
+
+                # Convert this number to the speed in rad/s per "percent speed"
+                radPerSecPerPercentSpeed = degPerSecPerPercentSpeed *\
+                    radiansPerDegree
```

Comment:
It seems like lines 140-191 could be moved up before the `while True` loop as most of these are either opening filehandles or defining constants.  Especially the part where we reset() the two motors and set the mode to run_direct()...does that really need to happen everytime through the loop?

Reply:
Ignore this, I see now (this isn't the balancing loop)

Reply:
Yep, there was a misleading comment that confused things. I've removed it.